raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansGrouper.java

```diff
@@ -86,6 +89,13 @@ StreamsBuilder buildTopologyWithClock(
                 Stores.persistentKeyValueStore(SPAN_STATE_STORE_NAME), keySerde, valueSerde)
             .withCachingEnabled();
 
+    StoreBuilder<KeyValueStore<IpIdentity, SpanMetadata>> ipsToSpanMetadataStateStoreBuilder =
```

Comment:
Correlation can be of anything, currently, we are thinking in terms of IPs but later it can take other forms say custom id etc. Think generic